[Ansible] Add splunk_config_override variable. (#950)
signalfx/splunk-otel-collector: deployments/ansible/roles/collector/README.md

  mode. (**default:** `/etc/otel/collector/agent_config.yaml` on Linux, 
  **default:** `%ProgramData%\Splunk\OpenTelemetry Collector\agent_config.yaml` on Windows)

- `splunk_config_override`: Custom Splunk OTel Collector config that will be merged into the default config.

> - `splunk_config_override_list_merge`: This variable is used to configure `list_merge` argument for `splunk_config_override`. When the hashes to merge contain arrays/lists. (**default:** `replace` the possible values are replace, keep, append, prepend, append_rp or prepend_rp)

Reviewer: ```suggestion
- `splunk_config_override_list_merge`: This variable is used to configure `list_merge` option for merging lists in `splunk_config_override` with lists in default config. Allowed options are values are `replace`, `keep`, `append`, `prepend`, `append_rp` or `prepend_rp`. More details: https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#combining-hashes-dictionaries. (**default:** `replace`)
```

Author: Updated as suggested.

  mode. (**default:** `/etc/otel/collector/agent_config.yaml` on Linux, 
  **default:** `%ProgramData%\Splunk\OpenTelemetry Collector\agent_config.yaml` on Windows)

> - `splunk_config_override`: Splunk OTel Collector config can be overridden with this variable. Additional components added using this variable will be merged into the default config.

Reviewer: ```suggestion
- `splunk_config_override`: Custom Splunk OTel Collector config can be merged into the default config.
```

Author: Okay, I will update accordingly.
Thanks.
